src/components/OwnerSettings.js

```diff
@@ -0,0 +1,120 @@
+import React, { useState, useEffect } from 'react';
+import { fetchStore, editStore, editSlot, fetchAllSlotsToday } from "../store"
+import { useDispatch, useSelector } from "react-redux"
+import { useParams } from "react-router-dom";
+import Grid from '@material-ui/core/Grid';
+import DateFnsUtils from '@date-io/date-fns';
+import {
+  MuiPickersUtilsProvider,
+  KeyboardTimePicker
+} from '@material-ui/pickers';
+import { makeStyles } from '@material-ui/core/styles';
+import InputLabel from '@material-ui/core/InputLabel';
+import MenuItem from '@material-ui/core/MenuItem';
+import Input from '@material-ui/core/Input';
+import FormControl from '@material-ui/core/FormControl';
+import Select from '@material-ui/core/Select';
+import history from "../utils/history";
+const moment = require("moment");
+
+const useStyles = makeStyles((theme) => ({
+  formControl: {
+    margin: theme.spacing(1),
+    minWidth: 120,
+  },
+  selectEmpty: {
+    marginTop: theme.spacing(2),
+  },
+}));
+
+function OwnerSettings(){
+    const store = useSelector(state => state.store);
+    const allSlotsToday = useSelector(state => state.allSlots);
+    let { storeId } = useParams();
+    const dispatch = useDispatch();
+    const classes = useStyles();
+    const [openingHour, setOpeningHour] = useState(new Date()); 
+    const [closingHour, setClosingHour] = useState(new Date());
+    const [maxPeoplePerSlot, setMaxPeoplePerSlot] = useState(0); 
+    const [slotDuration, setSlotDuration] = useState(20);
+
+    useEffect(() => {
+         dispatch(fetchStore(storeId))
+         dispatch(fetchAllSlotsToday(storeId));
+    }, [])
+
+    const isDifferent = (itemOne, itemTwo) => {
+        return itemOne != itemTwo ? true : false; 
+    }
+
+    const handleSubmit = e => {
+        e.preventDefault();
+        const openingInMilitaryTime = moment(openingHour).format("HH:mm");
+        const closingInMilitaryTime = moment(closingHour).format("HH:mm");
+        let editedSettings = { 
+            openingHour: openingInMilitaryTime, 
+            closingHour: closingInMilitaryTime, 
+            maxPeoplePerSlot: maxPeoplePerSlot, 
+            slotDuration: slotDuration 
+        }
+        dispatch(editStore(store, editedSettings));
+        const isNewMaxPeople = isDifferent(store.maxPeoplePerSlot, maxPeoplePerSlot);
+        if (isNewMaxPeople) {
+            allSlotsToday.map(slot => {
```

Comment:
Small but important, I think: if you're not interested in the new array that `.map` returns, `forEach` is more appropriate